src/shared/foundation-shared-components/components/agenda/FSAgendaHeader.vue

```diff
@@ -0,0 +1,188 @@
+        >
+          <FSSelectAgendaMode
+            :modelValue="$props.mode"
+            :hideHeader="true"
+            @update:modelValue="$emit('update:mode', $event)"
+          />
+          <FSButton
+            prependIcon="mdi-calendar-today-outline"
+            :label="$tr('ui.agenda.today', 'Today')"
+            @click="onToday"
+          />
+        </FSRow>
+      </FSRow>
+    </template>
+    <FSCol
+      v-else
+    >
+      <FSButton
+        prependIcon="mdi-calendar-today-outline"
+        :label="$tr('ui.agenda.today', 'Today')"
+        width="100%"
+        @click="onToday"
+      />
+      <FSRow
+        :wrap="false"
+        align="center-center"
+        padding="0 4px"
+      >
+        <FSButton
+          variant="icon"
+          icon="mdi-chevron-left"
+          @click="onPrevious"
+        />
+        <FSButton
+          width="100%"
+          color="primary"
+          :border="false"
+          :label="epochToDayMonthLongOnly($props.start)"
+          @click="showCalendarDialog = true"
+        />
+        <FSButton
+          variant="icon"
+          icon="mdi-chevron-right"
+          @click="onNext"
+        />
+      </FSRow>
+    </FSCol>
+  </FSRow>
+  <FSAgendaDialogCalendar
+    v-model:dialog="showCalendarDialog"
+    :modelValue="$props.start"
+    @update:modelValue="updateDateRange($event)"
+  />
+</template>
+
+<script lang="ts">
+import { defineComponent, type PropType, ref } from 'vue';
+
+import { useDateFormat } from '@dative-gpi/foundation-shared-services/composables';
+
+import { AgendaMode } from '@dative-gpi/foundation-shared-domain/enums/agendas';
+
+import FSCol from '../FSCol.vue';
+import FSRow from '../FSRow.vue';
+import FSButton from '../FSButton.vue';
+import FSSelectAgendaMode from './FSSelectAgendaMode.vue';
+import FSAgendaDialogCalendar from './FSAgendaDialogCalendar.vue';
+
+export default defineComponent({
+  name: 'FSAgendaHeader',
+  components: {
+    FSAgendaDialogCalendar,
+    FSSelectAgendaMode,
+    FSCol,
+    FSButton,
+    FSRow
+  },
+  props: {
+    mode: {
+      type: Number as PropType<AgendaMode>,
+      required: true
+    },
+    start: {
+      type: Number,
+      required: true
+    },
+    end: {
+      type: Number,
+      required: true
+    },
+    now: {
+      type: Number,
+      required: true
+    }
+  },
+  emits: ['update:start', 'update:end', 'update:mode'],
+  setup(props, { emit }) {
+    const { epochToMonthYearOnlyFormat, epochToDayMonthLongOnly, epochToLocalDayStart, epochToLocalDayEnd } = useDateFormat();
+
+    const showCalendarDialog = ref(false);
+
+    const updateDateRange = (dayStart: number) => {
+      const newStart = epochToLocalDayStart(dayStart);
+      if (props.mode === AgendaMode.Week) {
+        emit('update:start', newStart);
+        emit('update:end', epochToLocalDayEnd(newStart + (7 - new Date(newStart).getDay()) * 24 * 60 * 60 * 1000));
+      } else if (props.mode === AgendaMode.Month) {
```

Comment:
Pareil ici, emit() puis return comme ça ça évite les else if et tu mets un nouveau of par la ligne pour que ce soit plus lisible